pkg/wal/wal.go

```diff
@@ -0,0 +1,345 @@
+package wal
+
+import (
+	"context"
+	"fmt"
+	iradix "github.com/hashicorp/go-immutable-radix"
+	"go.uber.org/zap"
+	"io"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/segmentio/ksuid"
+
+	"github.com/oneconcern/datamon/pkg/storage"
+)
+
+type WAL struct {
+	mutableStore       storage.Store  // Location for updating token generator object
+	tokenGeneratorPath string        	// Path to the token generator object in the mutable store
+	walStore           storage.Store  // Append only store where WAL entries are written to
+	maxConcurrency     int           	// Max concurrency when reading
```

Comment:
I am not seeing clearly where this is used. I don't see anything with the go routines pattern that looks like a worker pool. Is that for the future?

Reply:
Next update has it.